docs/utilities/utilities.md

```diff
@@ -0,0 +1,232 @@
+---
+title: Utilities
+description: Utility
+---
+
+This module contains a set of utilities used internally in powertools and that you may also use in your Lambda functions.
```

Comment:
Since this is mostly internal utilities used by powertools, I think it would make sense to add a disclaimer here that some functionality might not be directly related to lambda functions and also can change without following semantic versioning. 

We would not want to block changes to this module until major version change if we really need to do some backward incompatible changes  or should we even document it to be used ?

Reply:
Actually, this is not true that it's internal, it is used with validation and idempotency modules. It was only used in validation before but can be used elsewhere now. I changed the sentence.